pkg/operator/ceph/object/user/controller.go

```diff
@@ -281,13 +281,59 @@ func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectSt
 		} else {
 			return errors.Wrapf(err, "failed to get details from ceph object user %q", u.Name)
 		}
+	} else if *user.MaxBuckets != *r.userConfig.MaxBuckets {
+		// TODO handle update for user capabilities
+		user, err = r.objContext.AdminOpsClient.ModifyUser(context.TODO(), *r.userConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create ceph object user %v", &r.userConfig.ID)
+		}
+		logCreateOrUpdate = fmt.Sprintf("updated ceph object user %q", u.Name)
+	}
+
+	var quotaEnabled = false
+	var maxSize int64 = -1
+	var maxObjects int64 = -1
+	if u.Spec.Quotas != nil {
+		if u.Spec.Quotas.MaxObjects != nil || u.Spec.Quotas.MaxSize != nil {
+			if (u.Spec.Quotas.MaxObjects != nil) && (*u.Spec.Quotas.MaxObjects != *user.UserQuota.MaxObjects) {
+				maxObjects = *u.Spec.Quotas.MaxObjects
+				quotaEnabled = true
+			}
+			if (u.Spec.Quotas.MaxSize != nil) && (u.Spec.Quotas.MaxSize.Value() != *user.UserQuota.MaxSize) {
+				maxSize = u.Spec.Quotas.MaxSize.Value()
+				quotaEnabled = true
+			}
+			if quotaEnabled {
+				userQuota := admin.QuotaSpec{
+					UID:        u.Name,
+					Enabled:    &quotaEnabled,
+					MaxSize:    &maxSize,
+					MaxObjects: &maxObjects,
+				}
+				err = r.objContext.AdminOpsClient.SetUserQuota(context.TODO(), userQuota)
+				if err != nil {
+					return errors.Wrapf(err, "failed to set quota for the user %v", u.Name)
+				}
+			}
+		} else if *user.UserQuota.Enabled {
+			userQuota := admin.QuotaSpec{
+				UID:        u.Name,
+				Enabled:    &quotaEnabled,
+				MaxSize:    &maxSize,
+				MaxObjects: &maxObjects,
+			}
+			err = r.objContext.AdminOpsClient.SetUserQuota(context.TODO(), userQuota)
+			if err != nil {
+				return errors.Wrapf(err, "failed to disable quota for the user %v", u.Name)
+			}
+		}
 	}
```

Comment:
We still need to be able to support the case where users remove quotas by removing the config from the yaml manifest entirely, setting `u.Spec.Quotas` back to `nil`. 

A code flow like below minimizes duplicate code. It could possibly be optimized, but it's usually best to start with simple code which is easier to read and understand. I find that especially for code that is idempotent, optimization can often introduce edge case bugs.

```go
var quotaEnabled = false
var maxSize int64 = -1 // negative means disabled, even if quota is enabled
var maxObjects int64 = -1 // negative means disabled, even if quota is enabled
if u.Spec.Quotas != nil {
		if u.Spec.Quotas.MaxObjects != nil {
			maxObjects = *u.Spec.Quotas.MaxObjects
			quotaEnabled = true
		}
		// NOTE: u.Spec.Quotas.MaxSize.Value() != *user.UserQuota.MaxSize is an optimization that no longer applies because we are always going to set the quota below, and it could introduce edge case bugs when a value doesn't change but quotas should still be enabled
		if u.Spec.Quotas.MaxSize != nil {
			maxSize = u.Spec.Quotas.MaxSize.Value()
			quotaEnabled = true
		}
}
// always set the quota so that it can be disabled if the user changes configs
userQuota := admin.QuotaSpec{
  UID:        u.Name,
  Enabled:    &quotaEnabled,
  MaxSize:    &maxSize,
  MaxObjects: &maxObjects,
}
err = r.objContext.AdminOpsClient.SetUserQuota(context.TODO(), userQuota)
if err != nil {
  // NOTE: changed %v to %q here to quote the name
  return errors.Wrapf(err, "failed to set quotas for user %q", u.Name)
}
```

Reply:
will modify according to this flow